src/gui_common/SegmentedBar.cs

```diff
@@ -0,0 +1,198 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Godot;
+
+/// <summary>
+///   A ProgressBar that is split up into IconProgressBars, data is stored in a dictionary
+/// </summary>
+public class SegmentedBar : ProgressBar
+{
+    public PackedScene IconProgressBarScene = GD.Load<PackedScene>("res://src/gui_common/IconProgressBar.tscn");
+    public string Type;
+
+    public void BarToggled(InputEvent @event, IconProgressBar bar)
+    {
+        if (@event is InputEventMouseButton eventMouse && @event.IsPressed())
+        {
+            IconBarConfig config = new IconBarConfig(bar);
+            config.Disabled = !config.Disabled;
+            new SegmentedBarConfig(this).handleBarDisabling(bar);
+        }
+    }
+}
+
+/// <summary>
+///   Configuration class for SegmentedBar class
+/// </summary>
+public class SegmentedBarConfig
+{
+    private SegmentedBar target;
+
+    public SegmentedBarConfig(SegmentedBar target)
+    {
+        this.target = target;
+    }
+
+    public float[] Size
+    {
+        get { return new float[]{ target.RectSize.x, target.RectSize.y }; }
+        set
+        {
+            target.RectSize = new Vector2(value[0], value[1]);
+            target.RectMinSize = target.RectSize;
+        }
+    }
+
+    public string Type
+    {
+        get { return target.Type; }
+        set { target.Type = value; }
+    }
+
+    public void updateAndMoveBars(Dictionary<string, float> data)
+    {
+        removeUnusedBars(target, data);
+
+        int location = 0;
+        foreach (var process in data)
+        {
+            createAndUpdateProcessBar(process, target, location);
+            location++;
+        }
+
+        foreach (var process in data)
+        {
+            if (target.HasNode(process.Key))
+                updateDisabledBars(process, target);
+        }
+
+        foreach (var process in data)
+        {
+            if (target.HasNode(process.Key))
+                moveBars(target.GetNode<IconProgressBar>(process.Key));
+        }
+    }
+
+    private void removeUnusedBars(ProgressBar parent, Dictionary<string, float> processes)
+    {
+        foreach (IconProgressBar progressBar in parent.GetChildren())
```

Comment:
This happens often, and the children are created in a controlled way. This causes an extra cast every iteration. The bars should instead be stored in a List to not have to cast these all the time, when it doesn't provide any benefit (it would be a breakage of encapsulation if any other code adds new bars).

Reply:
still a thing, also related to my previous comment.